oci/layout/oci_dest.go

```diff
@@ -144,6 +148,9 @@ func (d *ociImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo
 	if err := ensureParentDirectoryExists(blobPath); err != nil {
 		return types.BlobInfo{}, err
 	}
+
+	// need to explicitly close the file, since a rename won't otherwise not work on Windows
+	blobFile.Close()
```

Comment:
Per https://golang.org/pkg/io/#Closer

> The behavior of Close after the first call is undefined. 

i.e. the `defer … blobFile.Close()` must be deactivated, similarly to the `succeeded` boolean.

Reply:
hmm, one still would like defer to kick in until I can explicitly close. I'll take a look.